incubator/hnc/pkg/kubectl/configdelete.go

```diff
@@ -0,0 +1,61 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+var configDeleteCmd = &cobra.Command{
+	Use:   "delete-type --apiVersion X --kind Y",
+	Short: "Delete the HNC configuration of a specific type",
+	Example: fmt.Sprintf("  # Delete configuration of a core type\n" +
+		"  kubectl hns config delete-type --apiVersion v1 --kind Secret\n\n" +
+		"  # Delete configuration of a custom type\n" +
+		"  kubectl hns config delete-type --apiversion stable.example.com/v1 --kind CronTab"),
+	Run: func(cmd *cobra.Command, args []string) {
+		flags := cmd.Flags()
+		apiVersion, _ := flags.GetString("apiVersion")
+		kind, _ := flags.GetString("kind")
+		config := client.getHNCConfig()
+
+		i := 0
+		for _, t := range config.Spec.Types {
+			if t.APIVersion != apiVersion || t.Kind != kind {
+				config.Spec.Types[i] = t
+				i++
+			}
+		}
```

Comment:
I feel like the more Go-idiomatic way of doing this would be:

```
newTypes := []api.Whatever{}
found := false
for _, t := range config.Spec.Types {
  if t.APIVersion == apiVersion && t.Kind == kind {
    found := true
  } else {
    newTypes = append(newTypes, t)
  }
}
if !found { ... error ... }
config.Spec.Types = newTypes
```

The current way has a "loop" variable that isn't really incremented all the time plus you're modifying a list as you step over it, which I'm pretty sure works fine but always makes me closely re-read it to make sure I'm not missing something important.

Reply:
Thanks so much for the suggestion and explanation! Changed to more Go-idiomatic way.